src/main.rs

```diff
@@ -1,189 +1,169 @@
-                }
-                QueryResult::GetProviders(Err(err)) => {
-                    eprintln!("Failed to get providers: {:?}", err);
-                }
-                QueryResult::GetRecord(Ok(ok)) => {
-                    for PeerRecord { record: Record { key, value, .. }, ..} in ok.records {
-                        println!(
-                            "Got record {:?} {:?}",
-                            std::str::from_utf8(key.as_ref()).unwrap(),
-                            std::str::from_utf8(&value).unwrap(),
-                        );
-                    }
-                }
-                QueryResult::GetRecord(Err(err)) => {
-                    eprintln!("Failed to get record: {:?}", err);
-                }
-                QueryResult::PutRecord(Ok(PutRecordOk { key })) => {
-                    println!(
-                        "Successfully put record {:?}",
-                        std::str::from_utf8(key.as_ref()).unwrap()
-                    );
-                }
-                QueryResult::PutRecord(Err(err)) => {
-                    eprintln!("Failed to put record: {:?}", err);
-                }
-                QueryResult::StartProviding(Ok(AddProviderOk { key })) => {
-                    println!("Successfully put provider record {:?}",
-                             std::str::from_utf8(key.as_ref()).unwrap()
-                    );
-                }
-                QueryResult::StartProviding(Err(err)) => {
-                    eprintln!("Failed to put provider record: {:?}", err);
-                }
-                _ => {}
-            }
-            _ => {}
-        }
-    }
-}
+use libp2p::identify::{Identify, IdentifyConfig, IdentifyEvent};
+use async_std::channel::{unbounded, Sender};
+use libp2p::swarm::NetworkBehaviour;
+use std::time::Duration;
 
 #[async_std::main]
 async fn main() -> Result<(), Box<dyn Error>> {
-    println!("Welcome to rust coding with libp2p...Have fun!");
     env_logger::init();
 
     // Create a random key for ourselves.
     let local_key = identity::Keypair::generate_ed25519();
     let local_peer_id = PeerId::from(local_key.public());
 
-    println!("Peer ID {:?}", local_peer_id.to_string());
     // Set up a an encrypted DNS-enabled TCP Transport over the Mplex protocol.
-    let transport = development_transport(local_key).await?;
+    let transport = development_transport(local_key.clone()).await?;
+
     // We create a custom network behaviour that combines Kademlia and mDNS.
+    #[derive(NetworkBehaviour)]
+    #[behaviour(event_process = true)]
+    struct MyBehaviour {
+        kademlia: Kademlia<MemoryStore>,
+        identify: Identify,
+    }
+
+    impl NetworkBehaviourEventProcess<IdentifyEvent> for MyBehaviour {
+        fn inject_event(&mut self, event: IdentifyEvent) {
+
+            match event {
+                IdentifyEvent::Received {peer_id,info} => {
+                    println!("Connected peer_id: {:?}", &peer_id);
```

Comment:
[Nitpick] - Please remove the word `peer_id` since while printing we get 
`Connected peer_id: PeerId("12D3KooWPAMmexhwwjUEQJePyCrfJsLmrpTQw2jv4zhR5DyNbPcm")`
We do not need to show the word peer_id twice